cirq/google/known_devices.py

```diff
@@ -61,6 +61,16 @@ def __init__(self, constant: str, **kwargs) -> None:
     def __repr__(self):
         return self._repr
 
+    def _json_dict_(self):
+        return {
+            'cirq_type': self.__class__.__name__,
+            'constant': self._repr,
+            'measurement_duration': self._measurement_duration,
+            'exp_w_duration': self._exp_w_duration,
+            'exp_11_duration': self._exp_z_duration,
+            'qubits': sorted(self.qubits)
+        }
```

Comment:
When I went through with the original serialization PR, I took the opportunity to make slight modifications to classes to make all these classes follow more of a "dumb-container-of-data" model. The certification of a class is the ability to use `protocols.obj_to_dict_helper`. 

I think we should strongly consider

1. renaming `self._repr` to `self.constant`
2. exposing `_measurement_duration`, `_exp_w_duration`, `_exp_z_duration` as properties on `XmonDevice` *or* just rename those properties to not have the initial underscore.

@Strilanc : thoughts?

As a final note, I don't think we'd be able to get all the way because for json you have to do the conversion from qubits being a `frozenset` to a list